firmware/controllers/gauges/tachometer.cpp

```diff
@@ -9,47 +9,118 @@
  * @date Aug 18, 2015
  * @author Andrey Belomutskiy, (c) 2012-2018
  */
-
+//#include "engine.h"
 #include "tachometer.h"
 #include "trigger_central.h"
 
-#if !EFI_UNIT_TEST
 
 EXTERN_ENGINE;
 
-static scheduling_s tachTurnSignalOff;
+// [4 pulse/rev] * [2 edges per pulse] * [2 revs per cycle] = 16
+static scheduling_s events[16];
```

Comment:
looks like you want to say
#define MAX_TACH_PULSE 4

and use it to define array and also use it to validate user input?

what if user enters tachPulsePerRev =12?

Reply:
You are right, can we define maximum number of pulses supported? It would make sense no more then 12 pulses, since max 12 cylinders could be controlled, or if we take into consideration that 16 pistons are the max number that i know, maybe it would make sense to limit and add comment in TS?

Reply:
Of course - we can and should fire a warning if you try and use too many.  We can also set the limit in tunerstudio so that the user can't enter a number greater than 4.